crates/holochain_types/src/app.rs

```diff
@@ -12,6 +12,25 @@ pub type InstalledAppId = String;
 /// A friendly (nick)name used by UIs to refer to the Cells which make up the app
 pub type CellNick = String;
 
+/// The source of the DNA to be installed, either as binary data, or from a path
+#[derive(Clone, Debug, serde::Serialize, serde::Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum DnaSource {
+    /// register the dna loaded from a file on disk
+    Path(PathBuf),
+    /// register the dna as provided in the DnaFile data structure
+    Wasm(DnaFile),
```

Comment:
```suggestion
    DnaFile(DnaFile),
```
Maybe this might be less confusing? @guillemcordoba